### PR TITLE
LKE-7354: Refine space permissions

### PR DESCRIPTION
**Pull request by gmasclet**
https://linkurious.atlassian.net/browse/LKE-7354

As a user who is a part of a group with the _Manage Spaces_ right, I expect to see :
* All spaces on the _Resources_ page
* Only spaces that are shared with my group on the _Space dashboard_

### CODE REVIEW

File: src/api/spaces/index.ts [hunk 1]
```diff
@@ -52,30 +52,56 @@ export class SpacesAPI extends Request {
...
+      params: params
+    });
+  }
+}
+
+export class SpacesAPI extends Request {
+  /**
+   * List spaces visible by the current user.
+   */
+  public getSpaces(this: Request<ISpace[]>, params?: IDataSourceParams) {
```
Comment (Diachk) [inline src/api/spaces/index.ts:90]: I am not convinced that we need to separate name spaces for admin and regular users.
As of today the only difference in use-case is that:
- An admin want to see all spaces when managing them
- Regular users want to see the spaces shared with them in order too access the visualizations shared with them.

Proposal:
- Name the APIs in a way that reflects the use-cases:
1. getAllSpaces
2. getSpacesSharedWithMe

**Review by Diachk — CHANGES_REQUESTED** [review]
Review done.

There is an opportunity to simplify the spaces APIs by removing the APIs that we do not need and naming the existing Apis according to their use-case.

File: src/api/spaces/index.ts [hunk 2]
```diff
@@ -52,30 +52,56 @@ export class SpacesAPI extends Request {
...
       method: 'GET',
       params: params
     });
   }
 
   /**
-   * Get a space by id.
+   * Get a space by id (even if it is not visible by the current user).
    */
   public getSpace(this: Request<ISpace>, params: IGetSpaceParams) {
```
[the remark this answers is not in the document]
Comment (gmasclet) [reply src/api/spaces/index.ts]: Alright, done :ok_hand:

File: src/api/spaces/index.ts [hunk 3]
```diff
@@ -52,30 +52,56 @@ export class SpacesAPI extends Request {
...
+      url: '/:sourceKey/spaces',
+      method: 'GET',
+      params: params
+    });
+  }
+
+  /**
+   * Get a space by id (if it's visible by the current user).
+   */
+  public getSpace(this: Request<ISpace>, params: IGetSpaceParams) {
```
[the remark this answers is not in the document]
Comment (gmasclet) [reply src/api/spaces/index.ts]: Done :ok_hand:

File: src/api/spaces/index.ts (hunk 1 above)
Comment (gmasclet) [reply src/api/spaces/index.ts]: Let's do that, considering that me no longer have a "get space by id" operation. Having a distinct API for a single operation is definitely an overkill.

**Review by Diachk — APPROVED** [review]
LGTM

**Pull request merged**